docs/external-db.md

```diff
@@ -0,0 +1,159 @@
+# External DB integration
+
+Backstage hosts the data in the [PostgreSQL database](https://backstage.io/docs/getting-started/config/database/).
+By default, Operator creates and manages local instance of PostgreSQL in the same as Backstage deployment namespace but it also allows to switch this off and configure external instance instead.
+Usually, external connection requires more security, so, this instruction includes steps to configure SSL/TLS.
+
+### Configure your external PostgreSQL instance
+As a prerequisite you have to know:
+- **db-host** - your PostgreSQL instance DNS or IP address 
+- **db-port** - your PostgreSQL instance port number (usually 5432)
+- **username** - to connect to your PostgreSQL instance
+- **password** - to connect to your PostgreSQL instance
+
+In addition, to get your database connection secured with SSL/TLS you also need certificates in a form of PEM file. 
+
+You can find configuration guidelines for:
+- [AWS RDS PostgreSQL](#aws-rds-postgresql)
+- [Azure Database PostgreSQL](#azure-db-postgresql)
+
+### Create secret with PostgreSQL connection properties:
+````
+cat <<EOF | kubectl -n <your-namespace> create -f -
+apiVersion: v1
+kind: Secret
+metadata:
+ name: <cred-secret-name>
+type: Opaque
+stringData:
+ POSTGRES_PASSWORD: <password>
+ POSTGRES_PORT: <db-port>
+ POSTGRES_USER: <username>
+ POSTGRES_HOST: <db-host>
+ PGSSLMODE: require #  for TLS connection
```

Comment:
@gazarenkov Do your instructions work with this stricter mode?
If they do, I think we should recommend this since it provides both eavesdropping and MITM protection. See https://www.postgresql.org/docs/current/libpq-ssl.html#LIBPQ-SSL-PROTECTION
cc @kim-tsao

Reply:
it is the same in this case, "verify-full" implied if you configure certificate.
(I even think this env var is not mandatory in this case, you can try it yourself ;) )